src/app/pages/Swap/form/SwapFormInput/SwapInputHeader.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { memo, ReactNode } from 'react';
+
+import classNames from 'clsx';
+
+import { Button } from 'app/atoms';
+import Money from 'app/atoms/Money';
+import { T } from 'lib/i18n';
+import useTippy from 'lib/ui/useTippy';
+
+interface SwapInputHeaderProps {
+  label: ReactNode;
+  inputName: string;
+  isBalanceError: boolean;
+  assetDecimals: number;
+  handleSetMaxAmount: () => void;
+  assetBalanceStr?: string;
+}
+
+const SwapInputHeader = memo<SwapInputHeaderProps>(
+  ({ label, inputName, isBalanceError, assetDecimals, handleSetMaxAmount, assetBalanceStr }) => {
+    const fullBalanceStrRef = useTippy<HTMLButtonElement>({
+      trigger: 'mouseenter',
+      hideOnClick: false,
+      content: assetBalanceStr,
+      animation: 'shift-away-subtle'
+    });
+
+    return (
+      <div className="w-full flex items-center justify-between">
+        <span className="text-xs font-semibold text-text">{label}</span>
+        {assetBalanceStr && (
+          <span className="text-xs text-grey-1 flex items-baseline">
+            <span className="mr-1">
+              <T id="balance" />:
+            </span>
+            {inputName === 'input' ? (
```

Comment:
if someone change input name, i think it will be some time before testers notice the bug with setting max amount